src/ethereum/ethNetwork.js

```diff
@@ -741,25 +696,28 @@ export class EthereumNetwork {
       { tx: hexTx },
       baseUrl
     )
+    return this.broadcastResponseHandler(jsonObj, baseUrl, edgeTransaction)
+  }
 
-    if (typeof jsonObj.error !== 'undefined') {
-      this.ethEngine.error(
-        `FAILURE broadcastBlockCypher\n${JSON.stringify(
-          jsonObj.error
-        )}\n${cleanTxLogs(edgeTransaction)}`
-      )
-      throw jsonObj.error
-    } else if (jsonObj.tx && typeof jsonObj.tx.hash === 'string') {
+  broadcastResponseHandler(
+    res: JsonObject,
+    server: string,
+    tx: EdgeTransaction
+  ): BroadcastResults {
+    if (typeof res.error !== 'undefined') {
       this.ethEngine.error(
-        `SUCCESS broadcastBlockCypher\n${cleanTxLogs(edgeTransaction)}`
+        `FAILURE ${server}\n${JSON.stringify(res.error)}\n${cleanTxLogs(tx)}`
       )
+      throw res.error
+    } else if (typeof res.result === 'string') {
       // Success!!
-      return jsonObj
+      this.ethEngine.warn(`SUCCESS ${server}\n${cleanTxLogs(tx)}`)
```

Comment:
We're losing info on what type of broadcast function failed. Add as a parameter to broadcastResponseHandler